WordPress/Classes/Models/AbstractPost.m

```diff
@@ -22,7 +26,28 @@ - (void)awakeFromFetch
         // when wrong saving -- the app crashed for instance. So change our remote status to failed.
         [self setPrimitiveValue:@(AbstractPostRemoteStatusFailed) forKey:@"remoteStatusNumber"];
     }
+}
+
+- (void)setRemoteStatusNumber:(NSNumber *)remoteStatusNumber
+{
+NSLog(@"%@", NSStringFromSelector(_cmd));
+    self.metaIsLocal = ([remoteStatusNumber integerValue] == AbstractPostRemoteStatusLocal);
 
+    NSString *key = @"remoteStatusNumber";
+    [self willChangeValueForKey:key];
+    [self setPrimitiveValue:remoteStatusNumber forKey:key];
+    [self didChangeValueForKey:key];
+}
+
+- (void)setDate_created_gmt:(NSDate *)date_created_gmt
+{
+NSLog(@"%@", NSStringFromSelector(_cmd));
+    self.metaPublishImmediately = (date_created_gmt == nil);
+
+    NSString *key = @"date_created_gmt";
+    [self willChangeValueForKey:key];
+    [self setPrimitiveValue:date_created_gmt forKey:key];
+    [self didChangeValueForKey:key];
```

Comment:
Since you're supporting KVO, I recently learned that you should register dependent keys with `keyPathsForValuesAffecting*`

https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/KeyValueObserving/Articles/KVODependentKeys.html#//apple_ref/doc/uid/20002179-BAJEAIEE